### DISCUSSION TITLE
Is it safe to call realloc on a handle ?

### DISCUSSION THREAD

**Discussion by joy4eg** [libuv/libuv#4431] [Q&A] [answered]
Some code to start:
```c
uv_tcp_t *tcp;

tcp = malloc(sizeof(*tcp));
uv_tcp_init(loop, tcp);
uv_accept(tcp, server_handle);
// ... logic goes here ...
tcp = realloc(tcp, sizeof(*tcp) + size_of_additional_data) // <-- is it safe to do ?
```
The idea behind is to attach some additional data to the handle (like data buffers, context, etc), do some work with it, and free() everything together using one call. 

My main concern is that `realloc()` may return a new/different address after allocation, so I need to make sure that (old) address is not used anywhere in the `loop`.

**Comment by saghul** [comment, marked as answer]
The loop keeps track of all handles, so that's not going to work.

**Reply by joy4eg** [reply]
It's a pity :(
